jsoncppConfig.cmake.in

```diff
@@ -0,0 +1,16 @@
+cmake_policy(PUSH)
+cmake_policy(VERSION 3.0)
+
+@PACKAGE_INIT@
+
+include ( "${CMAKE_CURRENT_LIST_DIR}/jsoncpp-targets.cmake" )
+
+if(TARGET jsoncpp_static)
+    add_library(JsonCpp::JsonCpp ALIAS jsoncpp_static)
```

Comment:
I wish I understood why JsonCpp appears twice in this name.
https://cmake.org/cmake/help/latest/command/add_library.html?highlight=add_library#id6

We're making JsonCpp::JsonCpp an alias for whatever the TARGET is.  Why?

Reply:
CMake 3.0 target namespaces introduce a limited naming scope for the targets to avoid name clashes when targets named the same from different project or libraries. CMake recommends to put import targets of the library into appropriate namespace and I would like to use NAMESPACE feature of install(EXPORT) command to implement that. But it would break an existing target importing rules as JsonCpp already exports jsoncpp_static and jsoncpp_lib targets. 
Adding NAMESPACE directly to the install command would put them into namespace. An user would require to use JsonCpp::jsoncpp_static or JsonCpp::jsoncpp_lib to repair broken importing code.
To keep backward compatibility and improve a future user experience I conditionally add new ALIAS target that looks like namespaced target and may be used to import actual jsoncpp library that available.  
I assume the library is used usually as a source code to compile into the another project or a system level dependency in the form of static or shared library. Object files are not redistributable.
Although, static and shared library can be available at same time in the one build environment, it is rare and very specific case, as I know. So I would like to export an available library as an alias to use JsonCpp::JsonCpp generic namespaced library target and let CMake does all magic to set necessary compiler and linker flags. In such case projects that use JsonCpp::JsonCpp will able to use static or shared library depends on availability in build environment.